notebooks/day2/flatland.jl

```diff
@@ -0,0 +1,519 @@
+### A Pluto.jl notebook ###
+# v0.12.18
+
+using Markdown
+using InteractiveUtils
+
+# ╔═╡ 63f5861e-6244-11eb-268b-a16bc3f8265c
+using DSJulia, LinearAlgebra
+
+# ╔═╡ dfc779ee-6246-11eb-240b-4dc7a7d95641
+using Plots, RecipesBase
+
+# ╔═╡ b1d21552-6242-11eb-2665-c9232be7026e
+md"""
+# Flatland
+
+## Introduction and goal
+In this notebook, we will implement a variety of two-dimensional geometric shapes.
+The different shapes might have drastically different representations. For example, we can describe a rectangle
+by the coordinates of its center, its length and its width. A triangle, on the other hand,
+is more naturally represented by its three points. Similarly, computing the area of a rectangle or a triangle
+involves two different formulas. The nice thing about Julia is that you can hide this complexity from the users.
+You have to create your structures, subtypes of the abstract `Shape` type and have custom methods that will work
+for each type!
+
+Below, we suggest a variety of shapes, each with its unique representation. For this assignment, you have to complete **one**
+type and make sure all the provided functions `corners`, `area`, `move!`, `rotate!`,... work. Using `PlottingRecipes`, you can easily
+plot all your shapes (provided you implemented all the helper functions).
+
+Implementing such shapes can have various exciting applications, such as making a drawing tool or a ray tracer. Our
+end goal is to implement a simulator of a toy statistical physics system. Here, we simulate a system with inert particles, leading to self-organization.
+Our simple rejection sampling algorithm that we will use is computationally very demanding, an ideal case study for Julia!
+
+## Assignments
+
+- [ ] add the correct *inner* constructor to your type;
```

Comment:
Should I remove this since you show the constructors for each of these types?

Reply:
Removed some stuff, so this is an exercise